src/domain/transaction/evm/evm_transaction.rs

```diff
@@ -410,6 +445,22 @@ where
 
         let final_evm_data = updated_evm_data.with_signed_transaction_data(sig_result.into_evm()?);
 
+        // Validate the relayer has sufficient balance
+        let balance_validation = EvmTransactionValidator::validate_sufficient_relayer_balance(
+            bumped_price_params.total_cost,
+            &self.relayer().address,
+            &self.relayer().policies.get_evm_policy(),
+            &self.provider,
+        )
+        .await;
+
+        if let Err(validation_error) = balance_validation {
```

Comment:
Hey @LuisUrrutia 

Thanks for applying suggestion.

I think we are missing part where we exit flow after logging insufficient funds.

We do not want flow to continue in case of insufficient funds, but to exit from handler by returning.